File: report-generator/src/main/java/com/teamscale/report/util/ILogger.java
package com.teamscale.report.util;

/**
 * Minimal logging interface.
 * <p>
 * We use this to work around some strange problems when using log4j from the Teamscale Gradle plugin.
 */
public interface ILogger {

	/** Logs at debug level. */
	void debug(String message);

	/** Logs at info level. */
	void info(String message);

	/** Logs at warning level. */
	void warn(String message);

	/** Logs at warning level. */
	void warn(String message, Throwable throwable);

	/** Logs at error level. */
	void error(Throwable throwable);

	/** Logs at error level. The given {@link Throwable} may be null. */
	void error(String message, Throwable throwable);

	/** Logs at error level. */
	default void error(String message) {
		error(message, null);
	}
}

File: report-generator/src/main/kotlin/com/teamscale/report/util/ILogger.kt
package com.teamscale.report.util

/**
 * Minimal logging interface.
 *
 * We use this to work around some strange problems when using log4j from the Teamscale Gradle plugin.
 */
interface ILogger {

	/** Logs at debug level. */
	fun debug(message: String)

	/** Logs at info level. */
	fun info(message: String)

	/** Logs at warning level. */
	fun warn(message: String)

	/** Logs at warning level. The given [Throwable] may be null. */
	fun warn(message: String, throwable: Throwable?)

	/** Logs at error level. */
	fun error(throwable: Throwable)

	/** Logs at error level. The given [Throwable] may be null. */
	fun error(message: String, throwable: Throwable? = null)
}
